lib/phoenix/transports/long_poller/server.ex

```diff
@@ -103,6 +108,13 @@ defmodule Phoenix.Transports.LongPoller.Server do
   end
 
   @doc """
+  Detects disconnect broadcasts and shuts down
+  """
+  def handle_info(%Broadcast{event: "disconnect"}, state) do
+    {:stop, :normal, state}
```

Comment:
`{:stop, {:shutdown, :disconnected}, state}`

`:normal` is dangerous because links won't break.